api/app/src/external/commonwell/link.ts

```diff
@@ -0,0 +1,244 @@
+          const personLink = convertPersonToLink(person);
+
+          if (personLink) return personLink;
+        }
+      }
+    }
+  } catch (error) {
+    const msg = `Failure retrieving`;
+    console.log(`${msg} - link for person id:`, personId);
+    console.log(msg, error);
+    throw new Error(msg);
+  }
+};
+
+const createPatientLink = (personId: string, orgId: string, patientNumber: number): string => {
+  return `${apiUrl}/v1/person/${personId}/patientLink/${orgId}.${OIDNode.patients}.${patientNumber}${OID_URL_ENCODED_PREFIX}${orgId}/`;
+};
+
+export const findAllPersons = async (
+  patient: Patient,
+  organization: Organization
+): Promise<Link[]> => {
+  try {
+    const orgName = organization.data.name;
+    const orgId = organization.id;
+    const commonWell = makeCommonWellAPI(orgName, oid(orgId));
+
+    const cwPatientId = `${organization.id}.${OIDNode.patients}.${patient.patientNumber}^^^urn:oid:${organization.id}`;
+    const personsResp = await commonWell.searchPersonByPatientDemo(metriportQueryMeta, cwPatientId);
+
+    if (
+      personsResp &&
+      personsResp._embedded &&
+      personsResp._embedded.person &&
+      personsResp._embedded.person.length
+    ) {
+      return commonwellToLinks(personsResp._embedded.person);
+    }
+
+    return [];
+  } catch (error) {
+    const msg = `Failure retrieving persons`;
+    console.log(`${msg} - patient id:`, patient.id);
+    console.log(msg, error);
+    throw new Error(msg);
+  }
+};
+
+const commonwellToLinks = (persons: Person[]): Link[] => {
+  const links: Link[] = [];
+
+  persons.forEach(person => {
+    const personLink = convertPersonToLink(person);
+
+    if (personLink) links.push(personLink);
+  });
+
+  return links;
+};
+
+const convertPersonToLink = (person: Person): Link | null => {
+  const personId = getId(person);
+
+  const address = person.details?.address?.length ? person.details?.address[0] : undefined;
+  const personName = person.details?.name?.length ? person.details?.name[0] : undefined;
+
+  if (personId) {
+    const personLink: Link = {
+      entityId: personId,
+      potential: true,
+      source: LinkSource.commonWell,
+      patient: {
+        id: personId,
+        firstName: personName && personName.given?.length ? personName.given[0] : "",
+        lastName: personName && personName.family?.length ? personName.family[0] : "",
+        dob: person.details?.birthDate ? person.details.birthDate : "", // YYYY-MM-DD
+        gender: displayGender(person),
+        // TODO: ADD PERSON IDENTIFIERS
+        personalIdentifiers: [],
+        address: {
+          addressLine1: address && address.line ? address.line[0] : "",
+          city: address && address.city ? address.city : "",
+          state: address && address.state ? address.state : "",
+          zip: address && address.zip ? address.zip : "",
+          country: address && address.country ? address.country : "",
+        },
+        contact: {},
+      },
```

Comment:
This seems the opposite of `patientToCommonwell()`, right? Maybe we can move this to `external/commonwell/patient-conversion.ts` - and later I can move `patientToCommonwell()` there as well?